src/directory_sync/types/directory_user.rs

```diff
@@ -241,4 +250,69 @@ mod test {
             }
         )
     }
+
+    #[test]
+    fn it_returns_the_primary_email_when_the_user_has_a_primary_email() {
+        let directory_user = DirectoryUser {
+            id: DirectoryUserId::from("directory_user_01E1JG7J09H96KYP8HM9B0G5SJ"),
+            idp_id: "2836".to_string(),
+            directory_id: DirectoryId::from("directory_01ECAZ4NV9QMV47GW873HDCX74"),
+            organization_id: Some(OrganizationId::from("org_01EZTR6WYX1A0DSE2CYMGXQ24Y")),
+            username: Some("marcelina@foo-corp.com".to_string()),
+            emails: vec![DirectoryUserEmail {
+                primary: Some(true),
+                r#type: Some("work".to_string()),
+                value: Some("marcelina@foo-corp.com".to_string()),
+            }],
+            first_name: Some("Marcelina".to_string()),
+            last_name: Some("Davis".to_string()),
+            state: KnownOrUnknown::Known(DirectoryUserState::Active),
+            custom_attributes: HashMap::new(),
+            raw_attributes: RawAttributes(HashMap::new()),
+            timestamps: Timestamps {
+                created_at: Timestamp::try_from("2021-06-25T19:07:33.155Z").unwrap(),
+                updated_at: Timestamp::try_from("2021-06-25T19:07:33.155Z").unwrap(),
+            },
+        };
+
+        let primary_email = directory_user.primary_email();
+
+        assert_eq!(
+            primary_email,
+            Some(&DirectoryUserEmail {
+                primary: Some(true),
+                r#type: Some("work".to_string()),
+                value: Some("marcelina@foo-corp.com".to_string())
+            })
+        )
+    }
+
+    #[test]
+    fn it_returns_none_for_the_primary_email_when_the_user_does_not_have_a_primary_email() {
+        let directory_user = DirectoryUser {
+            id: DirectoryUserId::from("directory_user_01E1JG7J09H96KYP8HM9B0G5SJ"),
+            idp_id: "2836".to_string(),
+            directory_id: DirectoryId::from("directory_01ECAZ4NV9QMV47GW873HDCX74"),
+            organization_id: Some(OrganizationId::from("org_01EZTR6WYX1A0DSE2CYMGXQ24Y")),
+            username: Some("marcelina@foo-corp.com".to_string()),
+            emails: vec![DirectoryUserEmail {
+                primary: Some(false),
```

Comment:
Is it worth adding a test to cover the case when the `primary` field is `None`?

Reply:
It might be beneficial, but most likely only in the short-term.

I'd like to see this data structure get changed on the backend to ensure that the `primary` field always comes back. which would allow us to make this field a `bool` instead of an `Option<bool>`.